app/models/whitelist.rb

```diff
@@ -1,7 +1,37 @@
 class Whitelist
   include ActiveModel::Model
 
-  attr_accessor :email_domain, :organisation_name
+  attr_accessor :email_domain, :admin, :register, :organisation_name, :step
+
+  validate :validate_organisation_name, unless: proc { |whitelist|
+    whitelist.organisation_name.blank?
+  }
+  validate :validate_email_domain, unless: proc { |whitelist| whitelist.email_domain.blank? }
+
+  def next_step
+    { "first" => "first",
+      "second" => "second",
+      "third" => "third",
+      "fourth" => "fourth",
+      "fifth" => "fifth",
+      "sixth" => "sixth" }.fetch(step, "first")
+  end
```

Comment:
This looks odd to me, could it just be `step || "first"` or 

```
%w[first second third fourth fifth sixth].include?(step) ? step : "first"
```

Reply:
It is a bit odd I agree. I changed it to a set (which is backed by a hash), used a search similar to yours and used constants instead of strings